src/app/service/game-control.service.ts

```diff
@@ -77,7 +135,67 @@ export class GameControlService {
       .subscribe();
   }
 
+  reload() {
+    this.gameActive.next(true);
+    this.updateRound();
+    this.startWithCross = !this.startWithCross;
+    this.activePlayer.next(this.togglePlayer());
+    this.gameBoard.next(new Array(9).fill(Math.random(), 0, 9));
+    this.games$$.next(new ReplaySubject<move>());
+  }
+
+  private findWinner(moves: move[]) {
+    if (this.checkPattern(moves, 'circle')) {
```

Comment:
I would parametrize that method and move circle/cross to params.

Reply:
done